torch/prototypes/nestedtensor/codegen/tensorextension.py

```diff
@@ -0,0 +1,190 @@
+        'sign',
+        'sin',
+        'sinh',
+        'sqrt',
+        # 'sub',
+        'tan',
+        'tanh',
+        'tril',
+        'triu',
+        'trunc']
+
+
+def get_binary_functions():
+    return [
+        'add',
+        'mul',
+        'sub',
+        'div',
+    ]
+
+
+def get_comparison_functions():
+    return [
+        'eq',
+        'ge',
+        'gt',
+        'le',
+        'ne',
+    ]
+
+
+def set_function(module, cls, tfunc, func):
+    def _gen_func(tfunc):
+        orig_tfunc = getattr(torch, tfunc)
+
+        def _func(*args, **kwargs):
+            if isinstance(args[0], cls):
+                return func(*((tfunc, orig_tfunc,) + args), **kwargs)
+            else:
+                return orig_tfunc(*args, **kwargs)
+        return _func
+    setattr(module, tfunc, _gen_func(tfunc))
+
+
+def _check_meaningful_overwrite(cls, pbf):
+    class DefaultClass:
+        pass
+
+    if getattr(cls, pbf, False) and not getattr(DefaultClass, pbf, False):
+        raise Exception("WARNING: " + pbf + " already exists "
+                        "and not part of default class")
+
+
+def set_unary_method(cls, tfunc, pbf, inplace):
+    def _gen_func(pbf):
+        def _func(self: cls):
+            assert isinstance(self, cls)
+            if inplace:
+                return getattr(torch, tfunc)(self, out=self)
+            else:
+                return getattr(torch, tfunc)(self)
+        return _func
+    _check_meaningful_overwrite(cls, pbf)
+    setattr(cls, pbf, _gen_func(pbf))
+
+
+def set_binary_method(cls, tfunc, pbf, inplace):
+    def _gen_func(tfunc):
+        def _func(self: cls, other: cls):
+            if inplace:
+                return getattr(torch, tfunc)(self, other, out=self)
+            else:
+                return getattr(torch, tfunc)(self, other)
+        return _func
+    _check_meaningful_overwrite(cls, pbf)
+    setattr(cls, pbf, _gen_func(tfunc))
+
+
+def get_unary_method_signatures():
+    signatures = []
+    for pbf in get_unary_functions():
+        signatures.append({'torch': pbf, 'Tensor': pbf, 'inplace': False})
+        signatures.append({'torch': pbf, 'Tensor': pbf + "_", 'inplace': True})
+    return signatures
+
+
+def get_binary_method_signatures():
+    signatures = []
+    for pbf in ['add', 'mul', 'sub']:
+        signatures.append({'torch': pbf, 'Tensor': pbf, 'inplace': False})
+        signatures.append({'torch': pbf, 'Tensor': pbf + "_", 'inplace': True})
+        signatures.append({'torch': pbf, 'Tensor': "__" + pbf + "__", 'inplace': False})
+        signatures.append({'torch': pbf, 'Tensor': "__i" + pbf + "__", 'inplace': True})
+    signatures.append({'torch': 'div', 'Tensor': 'div', 'inplace': False})
+    signatures.append({'torch': 'div', 'Tensor': 'div' + "_", 'inplace': True})
+    signatures.append({'torch': 'div', 'Tensor': "__" + 'div' + "__", 'inplace': False})
+    signatures.append({'torch': 'div', 'Tensor': "__i" + 'div' + "__", 'inplace': True})
```

Comment:
Using a namedtuple would greatly improve readability here and also future-proof this from becoming like our ATen codegen